File: src/California.Creator.Service/Models/Core/LayoutAtom.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.Text;

namespace California.Creator.Service.Models.Core
{
    public class LayoutAtom : LayoutBase
    {
        public override int LayoutBaseId { get; set; } // overrides for client app code generation
        public override StyleMolecule StyleMolecule { get; set; }
        public override int LayoutSortOrderKey { get; set; } // default value => sql sequence
        public override LayoutType LayoutType { get { return LayoutType.Atom; } }

        public int PlacedAtomInBoxId { get; set; }
        public LayoutBox PlacedAtomInBox { get; set; }
        
        public ContentAtom HostedContentAtom { get; set; }

        public List<LayoutStyleInteraction> LayoutStyleInteractions { get; set; }

        [NotMapped] // TODO this property should only be visible client side
        public int Level { get; } = 0;
    }
}

File: src/California.Creator.Client/Typewriter/LayoutAtom.ts


import {CaliforniaProject} from "./CaliforniaProject";
import {ContentAtom} from "./ContentAtom";
import {LayoutBox} from "./LayoutBox";
import {StyleMolecule} from "./StyleMolecule";
import {LayoutType} from "./LayoutType";
import {LayoutStyleInteraction} from "./LayoutStyleInteraction"; 
export class LayoutAtom { 
    LayoutBaseId: number;
    StyleMolecule: StyleMolecule;
    LayoutSortOrderKey: number;
    LayoutType: LayoutType;
    PlacedAtomInBoxId: number;
    PlacedAtomInBox: LayoutBox;
    HostedContentAtom: ContentAtom;
    LayoutStyleInteractions: LayoutStyleInteraction[];
    Level: number;
}